pkg/object/equality_test.go

```diff
@@ -543,6 +543,21 @@ func TestPublisherProxyDeploymentEqual(t *testing.T) {
 			},
 			expectedResult: false,
 		},
+		"should be equal if replicas are the same": {
```

Comment:
should also be equal if the replicas are NOT the same, right?

Reply:
We changed the equality check to not take the replicas into consideration.

We have two tests:
- should be unequal if replicas changes
- should be equal if replicas are the same